lib/cn/cn_tree.c

```diff
@@ -2268,17 +2347,17 @@ cn_compact(struct cn_compaction_work *w)
      */
     sts_job_detach(&w->cw_job);
 
-    /* Commit the compaction if this isn't a spill. For a spill operation, each subspill to a child
-     * was committed as the spill progressed.
+    /* Commit the compaction if this isn't a spill.  For a spill operation,
+     * each subspill to a child was committed as the spill progressed.
      */
-    if (w->cw_action != CN_ACTION_SPILL)
+    if (w->cw_action != CN_ACTION_SPILL && w->cw_action != CN_ACTION_JOIN)
```

Comment:
I was going to update the comment to also talk about join, but I suspect @nabeelmmd will need to remove this join check...

Reply:
I will update the comment.. For join, I found it cleaner to do the cNDB commit within the cn_move() call itself.